proto/lib/config/deviceconfiguration.proto

```diff
@@ -24,4 +24,6 @@ message DeviceConfiguration {
     repeated ObservedFolder ignored_folders            = 14;
     repeated ObservedFolder pending_folders            = 15;
     int32                   max_request_kib            = 16 [(ext.goname) = "MaxRequestKiB", (ext.xml) = "maxRequestKiB", (ext.json) = "maxRequestKiB"];
+    bool                    show_web_address           = 17;
+    string                  web_address_port           = 18;
```

Comment:
Do we need a boolean to en-/disable? Alternative could be that empty `web_address_port` means disabled.

Also I think `web_ui_port` or `web_api_port` would be better names: It's only the port, not the address. And `web` alone is super generic, all the other ports (for protocol connections) also face the web. Probably even drop the `web` entirely.

Reply:
I think @AudriusButkevicius said there is a case where LAN have no port in the address, but I we definitely change the variable name. .https://github.com/syncthing/syncthing/pull/7017#issuecomment-701676367

Reply:
I think we need a global off switch, not a per device off switch. That off switch could simply be in javascript in local storage or something.

Reply:
Maybe even all of this could/should live in javascript only? Or do we expect any further usage outside of the web UI?

Reply:
Storing the configured port in the backend makes sense IMHO. This makes it accessible across different browsers and machines.